perun-core/src/test/java/cz/metacentrum/perun/core/api/AuthzResolverIntegrationTest.java

```diff
@@ -64,6 +64,378 @@ public void authorizedPerunAdmin() throws Exception {
+		AuthzResolver.setRole(sess, createdUser, createdVo, Role.TOPGROUPCREATOR);
+		AuthzResolver.setRole(sess, createdUser, createdVo, Role.VOOBSERVER);
+
+
+		AuthzResolver.refreshAuthz(session);
+		assertTrue(AuthzResolver.authorized(session, "test_voobserver_and_topgroupcreator", Arrays.asList(createdVo)));
+	}
+
+	@Test
+	public void authorizedCabinetAdmin() throws Exception {
+		final Vo createdVo = perun.getVosManager().createVo(sess, new Vo(0,"test123test123","test123test123"));
+
+		final Member createdMember = createSomeMember(createdVo);
+		final User createdUser = perun.getUsersManagerBl().getUserByMember(sess, createdMember);
+		PerunSession session = getHisSession(createdMember);
+		AuthzResolver.setRole(sess, createdUser, null, Role.CABINETADMIN);
+
+
+		AuthzResolver.refreshAuthz(session);
+		assertTrue(AuthzResolver.authorized(session, "test_cabinet", Arrays.asList()));
+	}
+
+	@Test
+	public void authorizedSelf() throws Exception {
+		final Vo createdVo = perun.getVosManager().createVo(sess, new Vo(0,"test123test123","test123test123"));
+
+		final Member createdMember = createSomeMember(createdVo);
+		final User createdUser = perun.getUsersManagerBl().getUserByMember(sess, createdMember);
+		PerunSession session = getHisSession(createdMember);
+
+		AuthzResolver.refreshAuthz(session);
+		assertTrue(AuthzResolver.authorized(session, "test_self", Arrays.asList(createdUser)));
+	}
+
+
+	@Test
+	public void authorizedSponsor() throws Exception {
+		final Vo createdVo = perun.getVosManager().createVo(sess, new Vo(0,"test123test123","test123test123"));
+
+		final Member createdMember = createSomeMember(createdVo);
+		final User createdUser = perun.getUsersManagerBl().getUserByMember(sess, createdMember);
+		PerunSession session = getHisSession(createdMember);
+
+		AuthzResolver.setRole(sess, createdUser, createdVo, Role.SPONSOR);
+
+		AuthzResolver.refreshAuthz(session);
+		assertTrue(AuthzResolver.authorized(session, "test_sponsor", Arrays.asList(createdVo,createdMember)));
+	}
+
+	@Test
+	public void authorizedResourceselfservice() throws Exception {
+		final Vo createdVo = perun.getVosManager().createVo(sess, new Vo(0,"test123test123","test123test123"));
+
+		final Member createdMember = createSomeMember(createdVo);
+		Group createdGroup = setUpGroup(createdVo, createdMember);
+		final User createdUser = perun.getUsersManagerBl().getUserByMember(sess, createdMember);
+		PerunSession session = getHisSession(createdMember);
+		AuthzResolver.setRole(sess, createdUser, createdGroup, Role.GROUPADMIN);
+
+		Facility facility = setUpFacility();
+		Resource resource = setUpResource(createdVo, facility);
+
+		perun.getResourcesManager().addResourceSelfServiceGroup(sess, resource, createdGroup);
+
+		AuthzResolver.refreshAuthz(session);
+		assertTrue(AuthzResolver.authorized(session, "test_resourceselfservice", Arrays.asList(resource)));
+	}
+
+	/*
+	@Test
+	public void authorizedServiceUser() throws Exception {
```

Comment:
You can erase this test for now. I think SERVICEUSER does not have any rights now.